### ISSUE TITLE
`FilePicker` times out when picking files on a slow network

### ISSUE THREAD

**Issue by getr11** [flet-dev/flet#771]
### Discussed in https://github.com/flet-dev/flet/discussions/738

<div type='discussions-op-text'>

<sup>Originally posted by **getr11** December 20, 2022</sup>
Hello, I am currently having a problem with the file-picker (_FilePickerResultEvent_). I tried the sample from here https://github.com/flet-dev/examples/blob/main/python/controls/file-picker/file-picker-upload-progress.py using flet==0.2.4 (Tried also the latest pre version). When I run the sample as _flet.WEB_BROWSER_ and select files from a _network share_ _FilePickerResultEvent_ randomly returns e.files is None even a file is selected. I can not reproduce this issue when running as _flet.FLET_APP_. It looks like a timing issue.
Has anyone experienced the same problem?</div>

**Comment by ndonkoHenri** [comment]
I havent experienced this yet.
"Files from a network share" ? 
Does this also happen with local files?

**Comment by alshapton** [comment]
Could it be a permissions issue ?
I dont know much about modern-day fileshares - however, could the browser be seeing the file as a local file in which case, could there be CORS issues from within the browser ?

**Comment by ndonkoHenri** [comment]
To be sure, try opening this stuff I made, then try uploading a text file.
You should be able to see the contents of the text file in the left area.
Let me know after you test please.

**Comment by getr11** [comment]
Strange phenomen. If you guys never experienced this it can still be an issue of the network i am connected. Unfortunately it's not possible to get the firewalls, ... turned off. Anyway it seems not to happen when accessing a local file. Today I made a few tests using some javascript doing actually the same. There was no problem. Then I went to flutter an give the package:file_picker/file_picker.dart a try. Same problem occurs here too. Do you know which flutter package flet uses for the FIlePicker? @ndonkoHenri  Can you attach the stuff you made for testings. Thanks a lot

**Comment by getr11** [comment]
I finally tried it on a very simple network at home (only a pc and a NAS). Could not reproduce this issue here. So I guess it might  be a timing issue or caused by firewall in the company network.

**Comment by ndonkoHenri** [comment]
Flet uses the package you mentioned above for the FliePicker: https://pub.dev/packages/file_picker

**Comment by ndonkoHenri** [comment]
> Can you attach the stuff you made for testings. 

I completely forgot that. Sorry.
Here it is: https://md-editor.fly.dev

**Comment by getr11** [comment]
Ok, found the problem. It's a problem of the file picker package [and](https://pub.dev/packages/file_picker) occurs when the network is slow. The picker detects that the filedialog is closed without a selection by a timeout of 500ms. Unfortunately my network sometimes takes more than 500ms for the signal that files are selected. Thus the picker returns with FileResult None because the timeout is reached. There is already an issue about the closing of the file picker open for the package I will go on there... Just one question about the flutter package sources used in flet. Is there any easy way to modify these directly?

That's the code part which cause the issue
`void cancelledEventListener(_) {
      window.removeEventListener('focus', cancelledEventListener);

      // This listener is called before the input changed event,
      // and the `uploadInput.files` value is still null
      // Wait for results from js to dart
      Future.delayed(Duration(milliseconds: 500)).then((value) {
        if (!changeEventTriggered) {
          changeEventTriggered = true;
          filesCompleter.complete(null);
        }
      });
    }`

**Comment by FeodorFitsner** [comment]
Maybe you could submit an issue in FilePicker repository?

**Comment by getr11** [comment]
Yes, I'll do. It's more about the architecture how this cancellation is implemented. Don't know if this can be changed easily. Anyway maybe they can provide the timeout as param to the pick files function....

**Comment by ndonkoHenri** [comment]
Upstream issue: https://github.com/miguelpruivo/flutter_file_picker/issues/1202

Closing this issue for now, while waiting for it to be fixed upstream.
